server/routes/userAPI.js

```diff
@@ -0,0 +1,34 @@
+const express = require("express");
+const db = require("../../db/models");
+
+const router = express.Router();
+
+// GET whether id is unique
+router.get("/api/user/:id/is_unique", (req, res) => {
+    db.User.find({ url: req.query.url })
+        .then((data) => {
+            if (data.length > 0) {
+                res.status(400).send(new Error("URL is already taken."));
+            } else {
+                res.status(200).send("URL is available!");
+            }
+        })
+        .catch((error) => {
+            console.log(error.message);
+            res.status(500).send(error);
+        });
+});
+
+// UPDATE user information
+router.post("/api/user/:id", (req, res) => {
+    // TODO read authentication
+
+    db.User.updateOne({ user_id: req.params.id }, { $set: req.body })
```

Comment:
This updates a user by matching with their id, correct? 
But what does `$set: req.body ` do?

Reply:
Haven't been able to test it in this instance yet, but it's mongoose's query updater! Hoping it will update any keys on the request body by passing in the whole thing. https://mongoosejs.com/docs/documents.html#updating-using-queries